ecommerce/src/main/java/com/algolia/instantsearch/examples/ecommerce/EcommerceActivity.java

```diff
@@ -79,6 +99,70 @@ protected void onDestroy() {
         refWatcher.watch(findViewById(R.id.hits));
     }
 
+    // endregion
+
+    // region Permission handling
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        if (requestCode == ID_REQ_VOICE_PERM && grantResults.length > 0) {
+            if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
```

Comment:
Feel free to ignore this comment, just suggesting in case you're interested: Something I've read and liked in the book `clean code` is to extract those kind of statements into small helper functions in order to have very readable code. 

We would get into something like: 
```
if (isVoicePermissionGranted(requestCode, grantResults) {
    showVoiceOverlay()
}
```
and the detailed code around checking for permission would be inside `isVoicePermissionGranted`. 

Again feel free to ignore, but just a small suggestion to make the code readable, especially when there s a lot of detailed stuff such as checking for length > 0  and using a lot of constants that are not interesting to know about for the reader of the code. Also could be applied to [this line](https://github.com/algolia/instantsearch-android-examples/pull/10/files#diff-56c398f95b5fef73a591e7517579a4fbR196) and some others.

Reply:
I like this a lot. This is especially easy to do with Kotlin extension functions, and would feel very natural. More on this later.